scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -62,35 +67,49 @@ abstract class ScanResultsStorage {
          * [FileBasedStorage] using a [XZCompressedLocalFileStorage] as backend is configured.
          */
         fun configure(config: ScannerConfiguration) {
-            val configuredStorages = listOfNotNull(
-                config.fileBasedStorage,
-                config.postgresStorage
-            )
+            storage = if (config.storages.isNullOrEmpty()) createDefaultStorage()
+            else createCompositeStorage(config)
+            log.info("ScanResultStorage has been configured to ${storage.name}.")
+        }
 
-            require(configuredStorages.size <= 1) {
-                "Only one scan results storage may be configured."
-            }
+        /**
+         * Create a [CompositeStorage] that manages all storages defined in the given [config].
+         */
+        private fun createCompositeStorage(config: ScannerConfiguration): ScanResultsStorage {
+            val storages = config.storages.orEmpty().mapValues { createStorage(it.value) }
 
-            when (val storageConfig = configuredStorages.singleOrNull()) {
-                null -> configureDefaultStorage()
-                is FileBasedStorageConfiguration -> configure(storageConfig)
-                is PostgresStorageConfiguration -> configure(storageConfig)
-                else -> throw IllegalArgumentException(
-                    "Unsupported configuration type '${storageConfig.javaClass.name}'."
-                )
-            }
+            fun resolve(name: String): ScanResultsStorage =
+                requireNotNull(storages[name]) {
+                    "Could not resolve storage '$name'."
+                }
+
```

Comment:
Did you consider that a `writer` may denote the exact same storage as a `reader`.
Am I right that in that case two instance would be created for the same storage.
I'm asking this having in mind potentially existing synchronisation mechanisms working on instance level.

Reply:
I believe that question is answered in the commit message of https://github.com/oss-review-toolkit/ort/pull/3155/commits/aa54709bd5ca94dbf084b7ca8cb2e75c80de8528, right?

Reply:
That's correct. If a storage is used as both reader and writer, it is configured once and then referenced as reader and writer; so it is the same instance.